lib/Page.js

```diff
@@ -257,6 +257,25 @@ class Page extends EventEmitter {
   }
 
   /**
+   * @param {string} username
+   * @param {string} password
+   * @return {!Promise}
+   */
+  async authenticate(username, password) {
+    return this._networkManager.authenticate(username, password);
+  }
+
+  /**
+   * @param {!Object} event
+   */
+  _onRequest(event) {
+    if (this.listenerCount(Page.Events.Request) > 0)
+      this.emit(Page.Events.Request, event);
```

Comment:
I might listen to the events without continuing them. This will hang the loading.